### PR TITLE
ISSUE-42: add continuous integration testing

### LINKED ISSUE #42: ISSUE-42: add continuous integration testing
Add CI to this project.  Build this project inside a docker container and use a service like codefresh to build that container, our code, and run unit tests.  Then make sure all of this is pulled into the pull request process.

### PR DESCRIPTION
**Pull request by godsflaw**
fixes #42 by adding a docker container that can be built and tested on codefresh.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +0,0 @@
-language: node_js
-node_js:
-  - '6'
-  - '5'
-  - '4'
-  - '0.12'
-  - '0.10'
-after_script:
-  - 'cat coverage/lcov.info | ./node_modules/.bin/coveralls'
```
Comment (godsflaw) [inline .travis.yml:9]: just don't need travis ci

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ADD src ${LOCALDIR}/src
+ADD test ${LOCALDIR}/test
+
+# install deps
+RUN apk upgrade --update && \
+  apk add --no-cache --virtual .gyp python make g++ opendkim-dev && \
+  npm install -g node-gyp
+
+# install codebase
+RUN (cd ${LOCALDIR} ; npm install ; node-gyp rebuild)
```
Comment (godsflaw) [inline Dockerfile:24]: need to look into why `npm install` doesn't trigger `node-gyp rebuild`

File: src/opendkim.h [hunk 3]
```diff
@@ -1,6 +1,10 @@
 #ifndef LIBOPENDKIM_H
 #define LIBOPENDKIM_H
 
+#if __GNUC__ > 4
+#define _Bool bool
+#endif
+
```
Comment (godsflaw) [inline src/opendkim.h:7]: This was tricky to track down.  In my development and production environments I am on `gcc` `4.x`, but in the docker container `gcc` was `6.x`.  This caused the following errors:
```
included from ../src/opendkim.h:11:0,
                 from ../src/opendkim.cc:1:
/usr/include/opendkim/dkim.h:456:30: error: '_Bool' has not been declared
 extern DKIM_STAT dkim_resign __P((DKIM *news, DKIM *olds, _Bool hdrbind));
[...]
```
I chased this down to a change in `stdbool.h` between versions and then made this macro.  I'm a little worried it was removed for a reason, but once I made this change, things compiled fine and the tests ran.  I'm open to suggestions on a better or more robust way to solve this.

**Comment by godsflaw** [comment]
Whoops, getting late, I wanted to configure codefresh to use the YAML file before review.

File: src/opendkim.h [hunk 4]
```diff
@@ -1,6 +1,10 @@
 #ifndef LIBOPENDKIM_H
 #define LIBOPENDKIM_H
 
+#if __GNUC__ > 4
+#define _Bool bool
```
Comment (chazomaticus) [inline src/opendkim.h:5]: ~Modern C has `<stdbool.h>` that does this for you--not sure if that's useful in this context.~ Nm, I see what this is doing now.

Comment (godsflaw) [reply src/opendkim.h:5]: this is what failed, the new header for gcc >4 doesn't work like the old one. :/

File: src/opendkim.h (hunk 3 above)
Comment (chazomaticus) [reply src/opendkim.h:7]: I think the "right" way to solve this is for dkim.h to only include `<stdbool.h>` if it's being compiled as C (not C++).  It looks like it has the `#ifdef __cplusplus; extern "C" { ...` trick, but I guess `stdbool.h` is missing checks to make it a no-op when compiled as C++, so dkim.h should do it instead.

Not sure there's anything you can do, and your solution seems fine to me!

**Review by chazomaticus — APPROVED** [review]
This looks great, glad it's working.  If you figure out what's up with `npm install` not doing the rebuild, that's awesome, but if not, there's nothing wrong with how it is now.  Ship it, and thanks!

**Pull request merged**